sql/src/main/java/io/crate/planner/fetch/FetchRewriter.java

```diff
@@ -122,6 +130,71 @@ private FetchDescription(TableIdent tableIdent,
         public TableIdent table() {
             return tableIdent;
         }
+
+        public Collection<? extends Symbol> preFetchOutputs() {
+            return preFetchOutputs;
+        }
+
+        /**
+         * Test if any Fields within {@code tree} are available pre-fetch.
+         *
+         */
+        public boolean availablePreFetch(Symbol tree) {
+            boolean[] available = new boolean[] { true };
```

Comment:
It's used in the function closure as a MutableBoolean. Could also use a AtomicRference or MutableBoolean from commons-lang3 if you prefer that